src/hsv.rs

```diff
@@ -53,87 +55,82 @@ impl<T:FloatChannel> Color<T> for Hsv<T> {
+    fn to_hsv<U:Channel>(&self) -> Hsv<U> {
         panic!("Not yet implemented")
     }
 }
 
-impl<T:Clone + FloatChannel> ToHsv for Hsv<T> {
+impl<T:Channel> ToHsv for Hsv<T> {
     #[inline]
-    fn to_hsv<U:FloatChannel>(&self) -> Hsv<U> {
-        Hsv::new(self.h.to_channel(),
+    fn to_hsv<U:Channel>(&self) -> Hsv<U> {
+        Hsv::new(Deg(cast(self.h.value())),
                  self.s.to_channel(),
                  self.v.to_channel())
     }
 }
 
-impl<T:Clone + FloatChannel> ToRgb for Hsv<T> {
+impl<T:Clone + Channel> ToRgb for Hsv<T> {
     fn to_rgb<U:Channel>(&self) -> Rgb<U> {
-        // Algorithm taken from the Wikipedia article on HSL and Hsv:
-        // http://en.wikipedia.org/wiki/HSL_and_Hsv#From_Hsv
-
-        let chr = self.v * self.s;
-        let h = self.h / cast(60u8);
-
-        // the 2nd largest component
-        let one: T = One::one();
-        let x = chr * (one - ((h % cast(2u8)) - One::one()).abs());
-
-        let mut rgb =
-            if      h < cast(1u8) { Rgb::new(chr.clone(), x, Zero::zero()) }
-            else if h < cast(2u8) { Rgb::new(x, chr.clone(), Zero::zero()) }
-            else if h < cast(3u8) { Rgb::new(Zero::zero(), chr.clone(), x) }
-            else if h < cast(4u8) { Rgb::new(Zero::zero(), x, chr.clone()) }
-            else if h < cast(5u8) { Rgb::new(x, Zero::zero(), chr.clone()) }
-            else if h < cast(6u8) { Rgb::new(chr.clone(), Zero::zero(), x) }
-            else                  { Rgb::new(Zero::zero(), Zero::zero(), Zero::zero()) };
-
-        // match the value by adding the same amount to each component
-        let mn = self.v - chr;
-
-        rgb.r = rgb.r + mn;
-        rgb.g = rgb.g + mn;
-        rgb.b = rgb.b + mn;
-
-        rgb.to_rgb::<U>()
+        if self.v.is_zero() {
+            rgb!(zero(), zero(), zero())
+        } else if self.s.is_zero() {
+            let gray = Channel::from(self.v);
+            rgb!(gray, gray, gray)
+        } else {
+            let max_f: f64 = cast(T::max()); 
+            let hue: f64 = cast(self.h.wrap().value());
+            let hue_six: f64 = hue / 360f64 * 6f64;
+            let hue_six_cat: usize = cast(hue_six);
+            let hue_six_rem: T = cast(hue_six.fract() * max_f);
+            let pv = Channel::from((T::max() - self.s).normalized_mul(self.v));
+            let qv = Channel::from((T::max() - self.s.normalized_mul(hue_six_rem)).normalized_mul(self.v));
+            let tv = Channel::from((T::max() - self.s.normalized_mul(T::max() - hue_six_rem)).normalized_mul(self.v));
+            let b: U = Channel::from(self.v);
+            match hue_six_cat {
+                0 | 6 => rgb!(b,tv,pv),
+                1 =>     rgb!(qv, b, pv),
+                2 =>     rgb!(pv, b, tv),
+                3 =>     rgb!(pv, qv, b),
+                4 =>     rgb!(tv, pv, b),
+                5 =>     rgb!(b, pv, qv),
+                _ => panic!("Unreachable code")
+            }
+        }
```

Comment:
no, i changed this to be able to do toRgb  conversion on non float hsv colors. i tried to adapt the original oiginal one but didn't manage to so i used this other variant of the algorithm.

not sure it's super useful since you usually won't have hsv images or something like that but i guess it doesn't hurt to have the option?